pkg/kp/rcstore/store.go

```diff
@@ -21,6 +21,19 @@ type Store interface {
 	Get(id fields.ID) (fields.RC, error)
 	List() ([]fields.RC, error)
 
+	// Watch for changes to the entire store and output the current list of RCs
+	// any time it changes. This function blocks; close the quit channel to
+	// terminate. Callers must drain the returned channels, including the error
+	// channel.
+	WatchNew(quit <-chan struct{}) (<-chan []fields.RC, <-chan error)
+
+	// Use the given session string (which uniquely identifies the lock holder)
+	// to acquire a lock on the specified replication controller. If the lock
+	// was successfully claimed, return (true, nil). If the request completed
+	// but the lock was already held, return (false, nil). Otherwise return an
+	// appropriate error.
+	Lock(id fields.ID, session string) (bool, error)
```

Comment:
I want the function to be as decoupled from the session as possible; therefore the caller instantiates the session and only they know what happens (ie keys are either deleted or released) when the session expires. I'll mention that the Lock key is ephemeral, therefore delete-on-expire sessions are safe to use.